### PR TITLE
Added Layer type

### PR DESCRIPTION
**Pull request by emreisikligil**
Layer type is added and used in Style objcet. This way, we can have
better validation and easier processing of layers.

### CODE REVIEW

File: styles.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ type Style struct {
 	// For example "mapbox://fonts/mapbox/{fontstack}/{range}.pbf"
 	Glyphs *string `json:"glyphs,omitempty"`
 	// Layers will be drawn in the order of this array.
-	Layers []json.RawMessage `json:"layers,omitempty"`
+	Layers []*Layer `json:"layers,omitempty"`
```
Comment (kszafran) [inline styles.go:70]: We normally use:
```suggestion
	Layers []Layer `json:"layers,omitempty"`
```

**Review by kszafran — CHANGES_REQUESTED** [review]
BTW, we should at some point just copy this whole repository into map-service. I don't see a reason to keep it in a repo of its own at this point.

**Review by gokhantamkoc — APPROVED** [review]
👍

File: styles.go (hunk 1 above)
Comment (emreisikligil) [reply styles.go:70]: In map-service, array of pointers is used for a similar structure. That's why I did it like that.

Comment (kszafran) [reply styles.go:70]: Hm.. yeah. Let's keep it as is for now.

**Review by kszafran — APPROVED** [review]

**Comment by emreisikligil** [comment]
> BTW, we should at some point just copy this whole repository into map-service. I don't see a reason to keep it in a repo of its own at this point.

It is cloned from another repo. That's why it's a separate repo.

**Comment by kszafran** [comment]
> > BTW, we should at some point just copy this whole repository into map-service. I don't see a reason to keep it in a repo of its own at this point.
> 
> It is cloned from another repo. That's why it's a separate repo.

I know. And that repo was created by our own guy :).

**Pull request merged**